### PR TITLE
Development v6 gh0.13

### PR DESCRIPTION
**Pull request by takb**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the development branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

### Information about the changes
- Reason for change:
update to GH 0.13

### CODE REVIEW

File: openrouteservice/src/main/java/org/heigit/ors/mapmatching/LocationIndexMatch.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public void setGpxAccuracy(double gpxAccuracyInMetern)
 
         for (int iteration = 0; iteration < 2; iteration++) {
             // should we use the return value of earlyFinish?
-            index.findNetworkEntries(queryLat, queryLon, set, iteration);
+                index.findNetworkEntries(queryLat, queryLon, set, iteration);
```
Comment (sfendrich) [inline openrouteservice/src/main/java/org/heigit/ors/mapmatching/LocationIndexMatch.java]: Is this additional indentation correct?

**Review by sfendrich — APPROVED** [review]
One minor comment.

File: openrouteservice/src/main/java/org/heigit/ors/mapmatching/LocationIndexMatch.java (hunk 1 above)
Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/mapmatching/LocationIndexMatch.java]: obviously not, fixed.

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java [hunk 2]
```diff
@@ -348,7 +349,7 @@ else if (ALT_ROUTE.equalsIgnoreCase(algoStr))
 
 			return altPaths;
 
-		} catch (IllegalArgumentException ex) {
+		} catch (Exception ex) {
```
Comment (rabidllama) [inline openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java]: Just as a clarification - do we want to be doing a catch-all Exception rather than more specific one?

**Review by rabidllama — COMMENTED** [review]
Looks good, there's one question about an Exception and when I know that I can approve.

Another thing though that we need to think about is with relation to coding style and if statements. In some files braces are used for if statements that only have one line, in others they are not. I guess that for this PR it is not so important as there is a mixed way of doing it in the files anyway, but we need to come up with a formalised style for these and then during code cleanup we can set everything to that method.

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ParamsTest.java [hunk 3]
```diff
@@ -588,8 +586,7 @@ public void expectCarToRejectWalkingAvoidables() {
 				.param("geometry", "true")
 				.param("profile", getParameter("carProfile"))
 				.param("options", options.toString())
-				.when()
-				.log().all()
+				.when().log().all()
```
Comment (aoles) [inline openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ParamsTest.java]: Why did you keep `log().all()` here but removed it from the other places?

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ResultTest.java [hunk 4]
```diff
@@ -836,27 +836,6 @@ public void testTollwaysExtraDetails() {
...
-				.body("routes[0].containsKey('extras')", is(true))
-				.body("routes[0].extras.tollways.values.size()", is(1))
-				.body("routes[0].extras.tollways.values[0][0]", is(0))
-				.body("routes[0].extras.tollways.values[0][1]", is(101))
-				.body("routes[0].extras.tollways.values[0][2]", is(0))
-				.statusCode(200);
-
-		checkExtraConsistency(response);
-
-		response = given()
```
Comment (aoles) [inline openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ResultTest.java:859]: Why was this test removed?? 😕

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java [hunk 5]
```diff
@@ -901,6 +893,17 @@ public void initLandmarkWeights(final int lmIdx, int lmNodeId, final long rowSiz
...
+        public void initFrom(int from, double weight){
+            super.initFrom(from, weight);
+        }
+
+        @Override
+        public void initTo(int to, double weight){
+            super.initTo(to, weight);
+        }
+
     }
```
Comment (aoles) [inline openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java:907]: Do we actually need to redefine the method in the subclass if it just forwards the call to the superclass? 🤔

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/HeavyVehicleFlagEncoder.java [hunk 6]
```diff
@@ -63,7 +65,7 @@ public HeavyVehicleFlagEncoder(PMap properties) {
         setBlockFords(properties.getBool("block_fords", true));
         setBlockByDefault(properties.getBool("block_barriers", true));
 
-        speedTwoDirections = properties.getBool("speed_two_directions", false);
+        speedTwoDirections = properties.getBool("speed_two_directions", true);
```
Comment (aoles) [inline openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/HeavyVehicleFlagEncoder.java:68]: Why has been the default changed from `false` to `true`? GH flag encoders seem to still fall back to `false`.

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/VehicleFlagEncoder.java [hunk 7]
```diff
@@ -49,7 +49,7 @@
 
     private double accelerationModifier = 0.0;
 
-    protected boolean speedTwoDirections = false;
+    protected boolean speedTwoDirections = true;
```
Comment (aoles) [inline openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/VehicleFlagEncoder.java]: Same here: Why the change?

**Review by aoles — CHANGES_REQUESTED** [review]
See the specific comments, thanks!

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java (hunk 2 above)
Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java]: no, I can't even tell you why that changed anymore... reverted.

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ParamsTest.java (hunk 3 above)
Comment (takb) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ParamsTest.java]: I got rid of all the then().log().all(), not the when().log().all(), since these tend to be longer, I can remove them entirely since the tests don't depend on them or anything.

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ResultTest.java (hunk 4 above)
Comment (takb) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ResultTest.java:859]: I don't know, some merge error on the way somewhere I guess...

Comment (takb) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ResultTest.java:859]: I'll put it back in

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java (hunk 5 above)
Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java:907]: yes, because those super methods are protected and we need to call them from here

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ParamsTest.java (hunk 3 above)
Comment (aoles) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/services/routing/ParamsTest.java]: No worries, I was just wondering why did you change them.

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/HeavyVehicleFlagEncoder.java (hunk 6 above)
Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/HeavyVehicleFlagEncoder.java:68]: because otherwise the graph doesn't build, we get an error " value for reverse direction would overwrite forward direction. Enable storeTwoDirections for this EncodedValue or don't use setReverse". I don't know if that property did use to get set somewhere, and that would be probably the proposed way, but on the other hand these FlagEncoder subclasses hard-code lots of profile-specific stuff anyway, so...

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/VehicleFlagEncoder.java (hunk 7 above)
Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/flagencoders/VehicleFlagEncoder.java]: this I just forgot to set back after I realised that didn't fix the HeavyVehicleFlagEncoder.

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java (hunk 2 above)
Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java]: and regarding the single conditional line braces, I'm all for deciding for one style and doing it everywhere.

**Review by aoles — APPROVED** [review]
LGTM! Thanks a lot for addressing the comments from my original review.
